### ISSUE TITLE
Gerrit: don't wipe Trybot-Result when updating commit messages

### ISSUE THREAD

**Issue by mvdan** [golang/go#30893]
For example, take https://go-review.googlesource.com/c/go/+/167978. I forgot to add a line to the commit message, so I updated it in patchset 2. That wouldn't affect the trybots at all, so I left them running for patchset 1 only.

However, that has the unfortunate side effect that I've now lost the Trybot-Result label on my CL. I could re-run the trybots, but that seems a bit wasteful.

It would be nice if the label wasn't wiped when just the commit message is updated. That is, assuming that the trybots are never going to care about the commit message. I don't think they do, right now, nor do I see a reason why they should.

If this is something Gerrit just isn't designed to do, perhaps we could just have Gobot post the Trybot-Result label again if a patchset only updated the commit message.

/cc @andybons @bradfitz @dmitshur

**Comment by dmitshur** [comment]
In what sense is it lost? I see a "TryBots are happy." message with a "TryBot-Result +1" vote at https://go-review.googlesource.com/c/go/+/167978#message-c36076efcbed3a38e87c5f6d4de374821854b7f5.

Are you referring to the top-level view of the CL not showing any votes?

<img width="248px" src="https://user-images.githubusercontent.com/1924134/54502668-35a0be80-4902-11e9-8fa4-b8a58aaadb88.png">

**Comment by bradfitz** [comment]
@dmitshur, yes, he's referring to the overall score on the CL. This is due to copyMinScore ... we have our Code-Review=+2 copied, but not +1 votes. See https://gerrit-review.googlesource.com/Documentation/config-labels.html#label_copyMinScore

But yes, the coordinator should & could check that a commit differs only in commit message and manually re-copy the vote without re-trying it. (But we shouldn't do the same with rebases, of course.)

**Comment by Gnouc** [comment]
@mvdan @bradfitz How about the case when the commit only update comment lines?

It's great if we can retain the result in this case, too.

**Comment by bradfitz** [comment]
@Gnouc, that's not 100% guaranteed to result in the same results. (gofmt checks, allhelp.go staleness checks, etc)

**Comment by dmitshur** [comment]
@heschik ran into this today in https://go-review.googlesource.com/c/go/+/174740/3#message-2ed0cb78dcc91bc5f91e7c424595f08bc212e465.

To avoid running into this again until this issue is resolved, don't update commit message right after starting a trybot run.

**Comment by dmitshur** [comment]
I'm planning to dedicate some time next week to this.

**Comment by dmitshur** [comment]
From looking into this, I understand this should be a matter of updating the [copyCondition](https://gerrit-review.googlesource.com/Documentation/config-labels.html#label_copyCondition) of the "Run-TryBot" and "TryBot-Result" labels to be NO_CODE_CHANGE (rather than its current NO_CHANGE):

> NO_CODE_CHANGE:
>
> Matches when a new patch set is uploaded that has the same parent tree as the previous patch set and the same code diff (including context lines) as the previous patch set. This means only the commit message may be different; the change hasn’t even been rebased. Also matches if the commit message is not different, which means this includes matching patch sets that have NO_CHANGE as the change kind.

Will mail a CL.

**Comment by dmitshur** [comment]
Implemented the change in https://go-review.googlesource.com/c/All-Projects/+/418075/. Editing only the commit message (but otherwise same parent and same tree) now keeps the Run-TryBot and TryBot-Result votes. So it's now possible to make commit message adjustments before submission without needing to rerun trybots.

(Coordinator still tracks the precise commit which it tested, so if the commit message is edited while a TryBot run is underway, the Run-TryBot vote will persist and coordinator will prefer to start again on the latest patch set. It's still better than before in that it restarts by default. Teaching coordinator to detect equal tree/parent and not restart is more involved and I'll consider out of scope of this issue.)
